robot/web/stream/server.go

```diff
@@ -191,13 +151,13 @@ func (srs *streamRPCServer) AddStream(ctx context.Context, req *streampb.AddStre
 				webrtc.PeerConnectionStateFailed,
 				webrtc.PeerConnectionStateClosed:
 
-				srs.ss.activeBackgroundWorkers.Add(1)
+				ss.activeBackgroundWorkers.Add(1)
 				utils.PanicCapturingGo(func() {
-					defer srs.ss.activeBackgroundWorkers.Done()
-					srs.ss.mu.Lock()
-					defer srs.ss.mu.Unlock()
-					defer delete(srs.ss.activePeerStreams, pc)
-					for _, ps := range srs.ss.activePeerStreams[pc] {
+					defer ss.activeBackgroundWorkers.Done()
+					ss.mu.Lock()
+					defer ss.mu.Unlock()
```

Comment:
I know this is how the original code worked, so just for my own edification:

Do we really want to hold this lock while `ps.stream.Stop()` called, or are we just trying to hold it while we mutate `ps.stream.Stop()`?

Reply:
It's unclear to me what the [concurrency contract](https://github.com/viamrobotics/rdk/blob/c629c2d3fc963777c719d0af3d6aa216929a482d/gostream/stream.go#L44-L45) is of `Stream.Stop`.

You know the implementing side of the interface a bit better than I do at the moment. Given the number of concurrency related smells, I'm inclined to leave this as-is for now. But I won't be surprised if you start making some drive-by changes in your PRs.